src/v0/destinations/hs/util.js

```diff
@@ -478,8 +487,16 @@ const getExistingData = async (inputs, destination) => {
   inputs.map(async (input) => {
     const { message } = input;
     const { destinationExternalId } = getDestinationExternalIDInfoForRetl(message, DESTINATION);
-    values.push(destinationExternalId);
+
+    if(typeof destinationExternalId === 'string'){
+      values.push(destinationExternalId.toLowerCase());
+    } else{
+      const value = typeof destinationExternalId === 'object' ? JSON.stringify(destinationExternalId).toLowerCase() : destinationExternalId.toString();
+      values.push(value);
+    }
```

Comment:
No need of this we would always expect external Id to be something simple like number or string.
Just do
```
values.push(destinationExternalId.toString().toLowerCase());
```

Reply:
updated